### PR TITLE
STR 731: Functional test RPC Behaviour when EL is down

### PR DESCRIPTION
**Pull request by voidash**
### Type of Change
Adds a functional test that checks the behavior of client RPC when reth is down and ability to continue when reth is up again

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactor
- [x] New or updated tests
- [ ] Dependency Update

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code where necessary.
- [x] I have updated the documentation if needed.
- [x] My changes do not introduce new warnings.
- [x] I have added tests that prove my changes are effective or that my feature works.
- [x] New and existing tests pass with my changes.

## Related Issues

STR-731

### CODE REVIEW

**Review by storopoli — DISMISSED** [review]
ACK e9cdf9d28a509446595c2ad57b89e408d4337b39

Nice work! I like the comments and the flow. It was easy to follow what was going on.

We need also an ACK from someone that has a solid knowledge of all the seq/reth RPCs that make sure that we are not missing anything.

File: functional-tests/fn_rpc_el_inactive.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        )
+        reth.stop()
+
+        assert not web3.is_connected(), "Reth did not stop"
+
+        # check if rpc is still working after sequencer has stopped
+        wait_until(
+            lambda: seqrpc.strata_protocolVersion() is not None,
+            error_with="Sequencer stopped after Reth stopped",
+        )
```
Comment (delbonis) [inline functional-tests/fn_rpc_el_inactive.py:53]: It shouldn't stop, it should just be reporting that we can't make new blocks.

File: functional-tests/fn_rpc_el_inactive.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        # check if sync status is working properly
+        wait_until(
+            lambda: seqrpc.strata_syncStatus() is not None,
+            error_with="Sequencer stopped after Reth stopped",
+        )
```
Comment (delbonis) [inline functional-tests/fn_rpc_el_inactive.py:59]: Same here.

File: functional-tests/fn_rpc_el_inactive.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+        # check if new l1 blocks are being recognized
+        cur_l1_height = seqrpc.strata_l1status()["cur_height"]
+        wait_until(
+            lambda: seqrpc.strata_l1status()["cur_height"] > cur_l1_height,
+            error_with="Sequencer stopped after Reth stopped",
+        )
```
Comment (delbonis) [inline functional-tests/fn_rpc_el_inactive.py:66]: This isn't the *sequencer* stopping if this stops happening.  Really it's the reader that would have crashed.

**Review by delbonis — CHANGES_REQUESTED** [review]

**Review by delbonis — APPROVED** [review]

**Comment by delbonis** [comment]
Any idea why these other functional tests are failing?

**Comment by storopoli** [comment]
> Any idea why these other functional tests are failing?

Flaky run. I've manually rerun and now passes.
I suspected a flaky run when I saw the error messages.

**Review by storopoli — APPROVED** [review]
ACK 4203132f08e1cc9e73b0d2f4c405b506fd9190dd

**Pull request merged**